Classes/Extensions/String+RegexSplitter.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  String+RegexSplitter.swift
+//  MobileOrg
+//
+//  Created by Mario Martelli on 12.12.16.
+//
+//
+//  This program is free software; you can redistribute it and/or
+//  modify it under the terms of the GNU General Public License
+//  as published by the Free Software Foundation; either version 2
+//  of the License, or (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+
+
+import Foundation
+
+public extension NSString {
+
+
+  func componentsSeparatedBy(regex: String) -> Array<String> {
+
+    guard let re = try? NSRegularExpression(pattern: regex, options: [])
+      else { return [] }
+
+    let stop = "<SomeStringThatYouDoNotExpectToOccurInSelf>"
+    let modifiedString = re.stringByReplacingMatches(
+      in: self as String,
+      options: [],
+      range: NSRange(location: 0, length: self.length),
+      withTemplate: stop) as NSString
+    return modifiedString.components(separatedBy: stop)
+  }
+
+
+  func arrayOfCaptureComponentsMatchedBy(regex: String) -> Array<Array<String>> {
+    let capture = self.captureComponentsMatchedBy(regex: regex)
+    if capture.count > 0 {
+      let ret:[Array<String>] = [capture]
+        return ret
+    }
+    return []
+  }
+
+  func captureComponentsMatchedBy(regex: String) -> Array<String> {
+
+    do {
+      var result:[String] = []
+      let swiftString = String(self)
+      let rgx = try NSRegularExpression(pattern: regex, options: [])
+
+      let matches = rgx.matches(in: swiftString, options: [], range: NSRange(location: 0, length: swiftString.characters.count))
+      for match in matches {
+        for n in 0..<match.numberOfRanges {
+          let range = match.rangeAt(n)
+          let begin = swiftString.index(swiftString.startIndex, offsetBy: range.location)
```

Comment:
This line consistently crashes for me with "fatal error: cannot increment beyond endIndex"